url/src/host.rs

```diff
@@ -172,6 +172,20 @@ impl<S: AsRef<str>> fmt::Display for Host<S> {
     }
 }
 
+impl<S, T> PartialEq<Host<T>> for Host<S>
+where
+    S: PartialEq<T>,
+{
+    fn eq(&self, other: &Host<T>) -> bool {
+        match (self, other) {
+            (Host::Domain(a), Host::Domain(b)) => *a == *b,
```

Comment:
(Unofficial review:) this looks good to me, although tiny nit: I think we can do without the `*` operators here, and that might look slightly nicer IMO.

Reply:
Done.